src/org/zaproxy/zap/extension/fieldenumeration/FieldEnumeration.java

```diff
@@ -0,0 +1,661 @@
+import java.awt.GridBagLayout;
+import java.awt.HeadlessException;
+import java.awt.EventQueue;
+import java.awt.Frame;
+import java.util.StringTokenizer;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.swing.*;
+import javax.swing.JTable;
+import javax.swing.AbstractAction;
+import javax.swing.table.DefaultTableModel;
+import javax.swing.event.CellEditorListener;
+import javax.swing.event.ChangeEvent;
+
+import org.apache.log4j.Logger;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVPrinter;
+import org.parosproxy.paros.view.View;
+import org.parosproxy.paros.db.DatabaseException;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.extension.AbstractDialog;
+import org.parosproxy.paros.model.HistoryReference;
+import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.model.Session;
+import org.zaproxy.zap.model.NameValuePair;
+import org.parosproxy.paros.network.HtmlParameter;
+import org.parosproxy.paros.extension.history.ExtensionHistory;
+import org.zaproxy.zap.model.SessionStructure;
+import org.zaproxy.zap.utils.ZapTextField;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpSender;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.view.messagecontainer.http.HttpMessageContainer;
+import org.zaproxy.zap.model.ParameterParser;
+import org.zaproxy.zap.view.widgets.WritableFileChooser;
+import org.parosproxy.paros.core.scanner.AbstractPlugin;
+
+public class FieldEnumeration extends AbstractDialog {
+
+	private static final long serialVersionUID = 1L;
+	private static final Logger LOGGER = Logger.getLogger(FieldEnumeration.class);	
+	private static final String CSV_EXTENSION = ".csv";
+
+	private HttpSender httpSender;
+	
+//	private final ExtensionHistory extension;
+        private JPanel jPanel = new JPanel();
+	private JTabbedPane tp = new JTabbedPane(); 
+	private JFrame result = new JFrame();
+	private JPanel formF = new JPanel();
+	private JPanel Panel = new JPanel();
+	private JLabel printURL = new JLabel();
+	private DefaultTableModel model = new DefaultTableModel();	
+	private JTable jTable = new JTable();
+	private JButton buttonOK = new JButton("SUBMIT");
+	private JComboBox<String> params = new JComboBox<String>();
+	private JComboBox<String> char_sets = new JComboBox<String>(); 
+	private String selected_chars  = null;
+	private JButton blacklist = new JButton("Blacklist");
+	private JButton save = new JButton("Export to CSV");
+	private GridBagConstraints c = new GridBagConstraints();
+	private GridBagConstraints c1 = new GridBagConstraints();
+	private JTextField preField = new JTextField();
+	private JTextField lowerR = new JTextField();
+	private JTextField highR = new JTextField();
+	private JTextField postField = new JTextField();
+	private Font font = new Font("Courier", Font.BOLD,12);
```

Comment:
Does this really need to be set?

This would override the users' UI settings selection.

Reply:
Is there any other solution?